### ISSUE TITLE
SVT : Mac, Eclipse, Appsody, project status transitions messed up.

### ISSUE THREAD

**Issue by mmulholl** [eclipse/codewind#499]
**Codewind version:**   Codewind tech preview	0.4.0.v201909191538	
**OS:** Mac

**Che version:**
**IDE extension version:**
**IDE version:** Eclipse committters 2019-06
**Kubernetes cluster:**

**Description:**

As an example a Appsody microprofile project status transition was
starting -> stopped (for 3 minutes) -> running.

Seen similar for  Appsody spring projects too

**Steps to reproduce:**

Create an Appsody microprofile project and watch.

**Workaround:** 

Don't give up too soon!

**Comment by eharris369** [comment]
Looks like it is related to the timeout.  I see this in the logs:
```
m[14/09/19 19:35:24 appsodyMP1] [INFO]9m Emitting event
 message: projectStatusChanged
 data: {
  "projectID": "c9236560-d726-11e9-a4c4-8f5d100c30de",
  "appStatus": "starting",
  "detailedAppStatus": "Pinging http://172.25.0.4:9080/health and http://172.25.0.4:9080/"
}
```
And then a few seconds later:
```
[32m[14/09/19 19:35:42 appsodyMP1] [INFO][39m Emitting event
 message: projectStatusChanged
 data: {
  "projectID": "c9236560-d726-11e9-a4c4-8f5d100c30de",
  "appStatus": "stopped",
  "appErrorStatus": "Failed to ping application due to timeout."
}
```

**Comment by eharris369** [comment]
Here is the full log: 
[logs.txt](https://github.com/eclipse/codewind/files/3632960/logs.txt)

**Comment by makandre** [comment]
^ Right, it's kind of what's expected now since the ping timeout was added in Codewind.  Unfortunately appsody projects hit this more often because the "run" phase actually includes the parts that download maven dependencies and builds the app.

**Comment by makandre** [comment]
@yangcao77 wondering is there any way to configure the ping timeout in Codewind?

**Comment by yangcao77** [comment]
@makandre  We currently hardcoded it to be 3 mins. @rajivnathan  I guess we want to make it configurable?

**Comment by rajivnathan** [comment]
@makandre Sounds like the "run" phase is doing too much? Wouldn't it be better to leave the app in building state if the app is still building? Once it completes then it can change state to Starting. Another reason we'd want to have it as part of the "build" phase is if there's a problem with the maven build, for example, then we'd want the status to be "Build failed", not "Failed to start" or equivalent. cc @elsony

**Comment by makandre** [comment]
Yes, that's ideal, but we currently don't have any means to find out what stage the `appsody run` command has reached (unless we do something like grepping the output which isn't a great solution)

**Comment by mmulholl** [comment]
Any plans to address this? Have codewind add some value by being intelligent rather than a pass through?

**Comment by makandre** [comment]
@mmulholl The timeout causing apps to go to `stopped` state is reverted in #1246.  So apps will stay in `starting` state until it goes to `running` state.  I think we can close this issue.

**Comment by yangcao77** [comment]
https://github.com/eclipse/codewind/issues/1246 only reverts the timeout change in 0.6.0. 
In post 0.6.0, we are also going remove the `stopped` state,  the apps will stay in `starting` state if it failed to ping. 
We will still keep and use the timeout to display warning in ide if the ping timeout is reached. (https://github.com/eclipse/codewind/issues/1297)
Also we will make the timeout configurable in `.cw-settings`, and increase the default timeout value for for appsody project. (https://github.com/eclipse/codewind/issues/1299)

**Comment by makandre** [comment]
@mmulholl #1246 is closed now so the status won't flip between `starting -> stopped -> running` anymore; it will just stay as `starting`.  Can we close this issue?  Thx.
